core/lib/env_config/src/contracts.rs

```diff
@@ -1,36 +1,30 @@
-use zksync_config::ContractsConfig;
+use zksync_config::{configs::EcosystemContracts, ContractsConfig};
 
 use crate::{envy_load, FromEnv};
 
+impl FromEnv for EcosystemContracts {
+    fn from_env() -> anyhow::Result<Self> {
+        Ok(Self {
+            bridgehub_proxy_addr: std::env::var("CONTRACTS_BRIDGEHUB_PROXY_ADDR")?.parse()?,
+            state_transition_proxy_addr: std::env::var("CONTRACTS_STATE_TRANSITION_PROXY_ADDR")?
+                .parse()?,
+            transparent_proxy_admin_addr: std::env::var("CONTRACTS_TRANSPARENT_PROXY_ADMIN_ADDR")?
+                .parse()?,
+        })
+    }
+}
+
 impl FromEnv for ContractsConfig {
     fn from_env() -> anyhow::Result<Self> {
         let mut contracts: ContractsConfig = envy_load("contracts", "CONTRACTS_")?;
-        // Note: we are renaming the bridge, the address remains the same
-        // These two config variables should always have the same value.
-        // TODO(EVM-578): double check and potentially forbid both of them being `None`.
-
-        contracts.l2_erc20_bridge_addr = contracts
-            .l2_erc20_bridge_addr
-            .or(contracts.l2_shared_bridge_addr);
-        contracts.l2_shared_bridge_addr = contracts
-            .l2_shared_bridge_addr
-            .or(contracts.l2_erc20_bridge_addr);
-
-        if let (Some(legacy_addr), Some(shared_addr)) = (
-            contracts.l2_erc20_bridge_addr,
-            contracts.l2_shared_bridge_addr,
-        ) {
-            if legacy_addr != shared_addr {
```

Comment:
We need these checks, did you maybe add them somewhere else? ( I'll try to find and close this comment)